analyzers/tests/SonarAnalyzer.UnitTest/Rules/CalculationsShouldNotOverflowSyntaxKindWalkerTest.cs

```diff
@@ -0,0 +1,241 @@
+            {
+                _ = i / i;
+            }
+            _ = i + i;
+        }
+        """, false, true)]
+    [DataRow("""
+        _ = i + i;
+        unchecked
+        {
+        }
+        """, true, true)]
+    [DataRow("""
+        unchecked
+        {
+            _ = i + i;
+            checked
+            {
+                unchecked
+                {
+                    _ = i + i;
+                }
+                _ = i / i;
+            }
+            _ = i + i;
+        }
+        """, false, true)]
+    [DataRow("""
+        unchecked
+        {
+            checked
+            {
+                unchecked
+                {
+                }
+                _ = i + i;
+            }
+        }
+        """, false, true)]
+    public void HasOverflowExpressions_Unchecked_Nesting(string statement, bool expectedHasOverflow, bool expectedIsUnchecked)
+    {
+        var tree = TestHelper.CompileCS(WrapInMethod($"""
+            var i = 0;
+            {statement}
+            """)).Tree;
+        var sut = new CalculationsShouldNotOverflow.SyntaxKindWalker();
+        sut.SafeVisit(tree.GetRoot());
+        sut.HasOverflow.Should().Be(expectedHasOverflow);
+        sut.IsUnchecked.Should().Be(expectedIsUnchecked);
+    }
+
+    [DataTestMethod]
+    [DataRow("_ = 1 + 1;", true)]            // Fix: should be false as it is checked at compile time
+    [DataRow("_ = int.MaxValue + 0;", true)] // OK: We would need to consult the semanticModel to learn that int.MaxValue is also a constant
+    [DataRow("_ = i + 1;", true)]
+    [DataRow("_ = 1 - 1;", true)]            // Fix: should be false as it is checked at compile time
+    [DataRow("_ = 1 * 1;", true)]            // Fix: should be false as it is checked at compile time
+    [DataRow("_ = i + 0;", true)]            // Fix: should be false as one operand is the identity element of addition
+    [DataRow("_ = i - 0;", true)]            // Fix: should be false as one operand is the identity element of subtraction
+    [DataRow("_ = i * 1;", true)]            // Fix: should be false as one operand is the identity element of multiplication
+    [DataRow("_ = i * 0;", true)]            // Fix: should be false as the result is always 0
+    [DataRow("""_ = "" + "";""", true)]      // Fix: should be false as both sides are string literals
+    [DataRow("""_ = s + "";""", true)]       // Fix: should be false as right side is a string literal
+    [DataRow("""_ = "" + s;""", true)]       // Fix: should be false as left side is a string literal
+    [DataRow("""_ = "" + s + s;""", true)]   // Fix: should be false as one of the operands is a string literal
+    [DataRow("""_ = s + "" + s;""", true)]   // Fix: should be false as one of the operands is a string literal
+    [DataRow("""_ = s + s + "";""", true)]   // Fix: should be false as one of the operands is a string literal
+    public void HasOverflowExpressions_Literals(string statement, bool expected)
```

Comment:
I would remove the `bool expected`, as it's always true. If `False` is needed, second method can be added, or parameter can be re-added later.

Reply:
This misses a bit the purpose of the tests: Enable an incremental optimization by turning these into `false`. I kept it this way.